src/features/workflow/WorkflowPage.tsx

```diff
@@ -58,7 +58,7 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({ qriRef }) => {
     dispatch(loadDataset(ref))
     dispatch(loadWorkflowByDatasetRef(qriRef))
 
-  }, [])
+  }, [ qriRef ])
```

Comment:
looks good, but a note I want to make sure you're aware of when passing a `qriRef` as a dep to hooks like `useEffect`: as written any change to the `qriRef` _object_ will cause a re-render, so if anything up the stack is, say, constructing the qri ref from URL params, that'll trigger a re-render. One way to avoid that is to spread the properties into the hook memoization array:

```tsx
const { username, name, path } = qriRef
useEffect(() => {
  const ref = newQriRef({ username, name, path})
  // ...
}, [username, name, path])
```

The way you have it written is totally fine, but if you find you're getting weird re-rendering, it's possible that the cause is reconstruction of a new qriRef object with the same properties. If that happens, switch to this form

Reply:
👍 I think this is actually cleaned up in some of my follow-on work, but I'll keep an eye out for it.